rust/src/schema.rs

```diff
@@ -100,6 +100,44 @@ impl Field {
         }
     }
 
+    /// Return Arrow Data Type name.
+    pub fn type_str(t: &DataType) -> String {
```

Comment:
Why do we need this method? Can we use `#Debug()` or `format` traits?

Reply:
It's used for keeping compatible with the names in [protobuf](https://github.com/eto-ai/lance/blob/414e953d415cc1941d8c743c36ab7119608fc394/rust/src/schema.rs#L68). I'm not sure if the results of `#Debug()` or `format` are compatible with the content from protobuf.

Reply:
https://docs.rs/arrow2/latest/arrow2/datatypes/enum.DataType.html#impl-Debug-for-DataType

What are the returns from here?

This implementation lacks support for quite some data types and causes panic in production code.

Reply:
The result is 
```
            println!("UInt8: {:?}",DataType::UInt8);
            println!("UInt16: {:?}",DataType::UInt16);
            println!("Binary: {:?}",DataType::Binary);
            println!("Extension: {:?}", DataType::Extension("the_str".to_string(), Box::new(DataType::Date64), Option::Some("option_string".into())));
-----
UInt8: UInt8
UInt16: UInt16
Binary: Binary
Extension: Extension("the_str", Date64, Some("option_string"))
```
I think what you said about panic is right but maybe we can start with simple cases first?

Reply:
Can we get a parity to https://github.com/eto-ai/lance/blob/ef3a53cfdedb39569b75e4c2343290452b2b8f30/cpp/src/lance/arrow/type.cc#L222 ?

These are the types we are currently using.

Reply:
Updated. Referred to [this](https://github.com/eto-ai/lance/blob/80cfd28bd0974a923366d6d3540fa0bfc4a66b35/cpp/src/lance/arrow/type.cc#L51), but the C++ version of Timestamp doesn't have a timezone field, while the rust version has one, for compatibility reasons I ignored the timezone field in rust.